azure-iot-device/azure/iot/device/iothub/abstract_clients.py

```diff
@@ -18,16 +18,16 @@
 class AbstractIoTHubClient(object):
     """A superclass representing a generic client. This class needs to be extended for specific clients."""
 
-    def __init__(self, transport):
+    def __init__(self, pipeline_adapter):
```

Comment:
From the perspective of the client, isn't this semantically a pipeline? The "adapter" bit is a detail of how the pipeline is provided, as I understand it.

Reply:
I have mixed feelings because I like it when the class name and the object name match, but your intuition feels correct to me so I'm going to change it.  Here and everywhere.

---
In reply to: [287146720](https://github.com/Azure/azure-iot-sdk-python-preview/pull/95#discussion_r287146720) [](ancestors = 287146720)